### PR TITLE
Gnome43/Libjpeg_turbo/ICU72.1

### LINKED ISSUE #7576: sommelier doesn't work properly on lenovo chromebook duet
## Description

After installing `sommelier` with chromebrew,  all GUI apps don't launch with the error `cannot open display`. I was trying to install `remmina` with the command,

`crew install remmina`

before launching remmina, I wanted to ensure if sommelier is working properly. so, I ran `restartsommelier` before launching remmina. but it gives the output, 

```
some sommelier processes failed to start
sommelier processes running:  
 
/usr/local/bin/startsommelier: line 39: return: can only `return' from a function or sourced script
```
and when I try to launch `remmina` with `G_MESSAGES_DEBUG=all`, my output was

```
** (process:26382): DEBUG: 23:26:28.707: (remmina_pref_init) - periodic_rmnews_last_get set to 0
** (process:26382): DEBUG: 23:26:28.715: (remmina_file_manager_init) - Initialized the "/usr/local/.config/.local/share/remmina" data folder
(process:26382): GLib-GIO-DEBUG: 23:26:28.720: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ?gio-vfs?
Load modules from /usr/local/lib/remmina/plugins
Remmina plugin glibsecret (type=Secret) has been registered, but is not yet initialized/activated. The initialization order is 2000.
** (process:26382): DEBUG: 23:26:28.862: (remmina_plugin_entry) - gfx_h264_available: 1
** (process:26382): DEBUG: 23:26:28.863: (remmina_rdp_settings_kbd_init) - rdp_keyboard_remapping_list: (null)
xprop:  unable to open display ':0'
xprop:  unable to open display ':0'
Failed to load plugin: /usr/local/lib/remmina/plugins/remmina-plugin-vnc.so.
Error: /usr/local/lib/libjpeg.so.9: version `LIBJPEG_8.0' not found (required by /usr/local/lib/libvncclient.so.1)
(process:26382): GLib-GIO-DEBUG: 23:26:28.877: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)

(process:26382): libsecret-CRITICAL **: 23:26:28.881: secret_service_load_collections_sync: assertion 'paths != NULL' failed
[glibsecret] unable to get secret service: Unknown error.
(process:26382): GLib-DEBUG: 23:26:28.882: unsetenv() is not thread-safe and should not be used after threads are created
(org.remmina.Remmina:26382): GLib-GIO-DEBUG: 23:26:28.883: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)

(org.remmina.Remmina:26382): Gtk-WARNING **: 23:26:28.886: cannot open display: :0
```
I have double checked my display variable. it is set to :0. 

Attempting to launch remmina with,

`sommelier -X --x-display=$DISPLAY --scale=1 remmina`

gives a different error, 

`error: failed to create virtwl context: Inappropriate ioctl for device`

I have tried reinstalling `remmina` with the `-s` option. It didn't change anything.
## Versions and system information

***
Output of `crew sysinfo`

```
/usr/local/bin/crew:1820:in `sysinfo_command': uninitialized constant ARCH_ACTUAL (NameError)

    - Architecture: `#{ARCH_ACTUAL}` (`#{ARCH}`)
                       ^^^^^^^^^^^
        from /usr/local/bin/crew:1867:in `<main>'
```
Output of `cat /etc/lsb-release`

```
CHROMEOS_ARC_ANDROID_SDK_VERSION=28
CHROMEOS_ARC_VERSION=9138510
CHROMEOS_AUSERVER=https://tools.google.com/service/update2
CHROMEOS_BOARD_APPID={50F3C95B-CA5B-4AF8-87A2-8CD19588BD12}
CHROMEOS_CANARY_APPID={90F229CE-83E2-4FAF-8479-E368A34938B1}
CHROMEOS_DEVSERVER=
CHROMEOS_RELEASE_APPID={50F3C95B-CA5B-4AF8-87A2-8CD19588BD12}
CHROMEOS_RELEASE_BOARD=kukui-signed-mp-v4keys
CHROMEOS_RELEASE_BRANCH_NUMBER=87
CHROMEOS_RELEASE_BUILDER_PATH=kukui-release/R107-15117.87.0
CHROMEOS_RELEASE_BUILD_NUMBER=15117
CHROMEOS_RELEASE_BUILD_TYPE=Official Build
CHROMEOS_RELEASE_CHROME_MILESTONE=107
CHROMEOS_RELEASE_DESCRIPTION=15117.87.0 (Official Build) stable-channel kukui 
CHROMEOS_RELEASE_KEYSET=mp-v4
CHROMEOS_RELEASE_NAME=Chrome OS
CHROMEOS_RELEASE_PATCH_NUMBER=0
CHROMEOS_RELEASE_TRACK=stable-channel
CHROMEOS_RELEASE_UNIBUILD=1
CHROMEOS_RELEASE_VERSION=15117.87.0
DEVICETYPE=CHROMEBOOK
GOOGLE_RELEASE=15117.87.0
```
***

Any help is much appreciated. Thanks !

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #7576

- Trying to get webkit2gtk 2.38.1 to build for `armv7l`... (This is the major holdup remaining )
- Switched all packages to `libjpeg-turbo` from `libjpeg`. (Recent versions of standalone libjpeg have made incompatible API changes which are not widely supported.)
- `icu4c` -> 72.1
- PHP apps MAY need rebuilding since they are built against the old libjpeg. (@uberhacker if there are new versions which need building, maybe we just queue those up?)
- Use `CREW_CACHE_BUILD=1  CREW_CACHE_FAILED_BUILD=1` if needed for builds.

- Packages left to be built ( mostly connected to the armv7l `webkit2gtk` build issue ):
  - `webkit2gtk_4` [ soup2+, gtk4-] (  needs build for `armv7l`)
  - `webkit2gtk_4_1` [soup2-, gtk4-] ( needs build for `armv7l`, `i686` )
  - `webkit2gtk_5` [soup2-, gtk4+] ( needs build for `armv7l`, `i686` )
  - `gemacs` ( depends on `webkit2gtk_4`, needs build for `armv7l` )
  - `evolution_data_server` ( depends on `webkit2gtk_4_1` and `webkit2gtk_5`, needs build for `armv7l` )
  - `epiphany` ( depends on `webkit2gtk_4_1`, needs build for `armv7l` )

_Builds_ properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l`

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=gnome43_libjpeg_turbo_icu72 CREW_TESTING=1 crew update ; yes | crew upgrade
```

### CODE REVIEW

**Comment by satmandu** [comment]
Aside from the packages still building mentioned above, this PR is ready for review.

**Comment by satmandu** [comment]
Added the changes from the crew cache PR so I can use it during the webkit2gtk builds...

**Comment by satmandu** [comment]
Waiting for webkit2gtk 2.38.1 builds...

**Comment by satmandu** [comment]
All packages should be up to date, and WIP webkit2gtk packages left as is for the moment until we can figure out build fixes from upstream.

File: packages/devil.rb [hunk 1]
```diff
@@ -3,26 +3,21 @@
...
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/devil/1.8.0-e342_i686/devil-1.8.0-e342-chromeos-i686.tar.xz',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/devil/1.8.0-e342_x86_64/devil-1.8.0-e342-chromeos-x86_64.tar.xz'
+    x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/devil/1.8.0-6f3d5e9_x86_64/devil-1.8.0-6f3d5e9-chromeos-x86_64.tar.zst'
   })
   binary_sha256({
-    aarch64: '7d45e28698eb99159d98b97760d11ddecff0ac87412b7cf249fb7e188b320502',
-     armv7l: '7d45e28698eb99159d98b97760d11ddecff0ac87412b7cf249fb7e188b320502',
-       i686: 'c75491427f9a75e33b8bca9f33b24cef15afc869118298d41177d45d3a913f4b',
-     x86_64: '99de10866bec2ac80094d5f3ad8a4d906cb6597dd01a2cce7d408533488ce330'
+    x86_64: '494336d49fee91201c74574356a2bf5c1e5bdc6e5d15bcc03decb99210286664'
```
Comment (uberhacker) [inline packages/devil.rb:16]: Missing builds for arm and i686?

File: packages/evolution_data_server.rb [hunk 2]
```diff
@@ -2,32 +2,48 @@
...
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/evolution_data_server/3.40.3_x86_64/evolution_data_server-3.40.3-chromeos-x86_64.tpxz'
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/evolution_data_server/3.46.0_i686/evolution_data_server-3.46.0-chromeos-i686.tar.zst',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/evolution_data_server/3.46.0_x86_64/evolution_data_server-3.46.0-chromeos-x86_64.tar.zst'
   })
   binary_sha256({
-    aarch64: '8ade9630fc18351495084f5e0ae3ad98fbbcd3e50f937d795d8dc2a1c4c36158',
-     armv7l: '8ade9630fc18351495084f5e0ae3ad98fbbcd3e50f937d795d8dc2a1c4c36158',
-     x86_64: '9620cad89b0ed797c84b4d7b85b977883678669963c724a679a09c681ef3934f'
+       i686: '303248f7eb4e050cbf846caaf9f23b04c19df4fa624ee90db6c799f891175271',
+     x86_64: '57189bbc3c4fa355db5d3e64dc89521f91e7366eadb251b993bd963ec573d1d3'
```
Comment (uberhacker) [inline packages/evolution_data_server.rb:18]: Missing arm build?

File: packages/gemacs.rb [hunk 3]
```diff
@@ -3,47 +3,67 @@
...
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gemacs/28.2_i686/gemacs-28.2-chromeos-i686.tar.zst',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gemacs/28.2_x86_64/gemacs-28.2-chromeos-x86_64.tar.zst'
   })
   binary_sha256({
-    aarch64: '8ba0433113a23361cf961d372c8e7db79206c5baf3046255bebec0acd1d73452',
-     armv7l: '8ba0433113a23361cf961d372c8e7db79206c5baf3046255bebec0acd1d73452',
-       i686: 'e1a0efa63cf40683ddfd90ca181f43e50d87bbf6179774f976e6cc8f917f0205',
-     x86_64: '8f462f3de1a19f67a36e6752124da213add709a075cd8bc52815452d9bf68b16'
+       i686: 'd3c1c2133ca735512de8b77069093f78110a41377962080a69271d5052d18a46',
+     x86_64: 'ed1a8eacde7a8a17be9fb11a2bcdf83d3ce03ba3d78ebbb71884f808f515debd'
```
Comment (uberhacker) [inline packages/gemacs.rb:18]: Missing arm build?

File: packages/gnome_maps.rb [hunk 4]
```diff
@@ -44,8 +56,4 @@ def self.build
   def self.install
     system "DESTDIR=#{CREW_DEST_DIR} ninja -C builddir install"
   end
-
-  def self.postinstall
-    system "glib-compile-schemas #{CREW_PREFIX}/share/glib-2.0/schemas"
-  end
```
Comment (uberhacker) [inline packages/gnome_maps.rb:50]: Why was this removed?

File: packages/gstreamer.rb [hunk 5]
```diff
@@ -3,32 +3,108 @@
 class Gstreamer < Package
   description 'GStreamer is a library for constructing graphs of media-handling components.'
   homepage 'https://gstreamer.freedesktop.org/'
-  @_ver = '1.20.0'
-  version @_ver
+  @_ver = '1.20.3'
+  version "#{@_ver}-1"
```
Comment (uberhacker) [inline packages/gstreamer.rb:7]: `-1` not necessary?

Comment (satmandu) [reply packages/gstreamer.rb]: fixed.

File: packages/gemacs.rb (hunk 3 above)
Comment (satmandu) [reply packages/gemacs.rb:18]: depends upon `webkit2gtk_4`, which isn't building for `armv7l`.

File: packages/evolution_data_server.rb (hunk 2 above)
Comment (satmandu) [reply packages/evolution_data_server.rb:18]: depends upon webkit2gtk_4_1, which isn't building for armv7l.

File: packages/devil.rb (hunk 1 above)
Comment (satmandu) [reply packages/devil.rb:16]: Builds for those architectures are currently broken, as devil has not been updated to work with newer versions of jasper.

See https://github.com/DentonW/DevIL/pull/102

File: packages/mpv.rb [hunk 6]
```diff
@@ -4,21 +4,23 @@ class Mpv < Package
...
-  compatibility 'x86_64 armv7l aarch64'
+  compatibility 'armv7l aarch64 x86_64'
   source_url 'https://github.com/mpv-player/mpv.git'
   git_hashtag "v#{@_ver}"
 
   binary_url({
-    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mpv/0.34.1_armv7l/mpv-0.34.1-chromeos-armv7l.tar.zst',
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mpv/0.34.1_armv7l/mpv-0.34.1-chromeos-armv7l.tar.zst',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mpv/0.34.1_x86_64/mpv-0.34.1-chromeos-x86_64.tar.zst'
+    i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mpv/0.34.0_i686/mpv-0.34.0-chromeos-i686.tpxz',
```
Comment (uberhacker) [inline packages/mpv.rb:14]: Version doesn't match?

File: packages/pygobject.rb [hunk 7]
```diff
@@ -3,31 +3,34 @@
 class Pygobject < Package
   description 'PyGObject is a Python package which provides bindings for GObject based libraries such as GTK+, GStreamer, WebKitGTK+, GLib, GIO and many more.'
   homepage 'https://pygobject.readthedocs.io/'
-  @_ver = '3.42.0'
-  version "#{@_ver}-1"
+  @_ver = '3.42.2'
+  version @_ver.to_s
```
Comment (uberhacker) [inline packages/pygobject.rb:7]: Can we get rid of this ridiculous rubocop `.to_s` madness?

Comment (satmandu) [reply packages/pygobject.rb]: Changed in all packages.

File: packages/mpv.rb (hunk 6 above)
Comment (satmandu) [reply packages/mpv.rb:14]: Accidental build failure. I'll start a rebuild for i686...

File: packages/webkit2gtk_4.rb [hunk 8]
```diff
@@ -1,133 +1,180 @@
...
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/webkit2gtk_4/2.38.0-1_i686/webkit2gtk_4-2.38.0-1-chromeos-i686.tar.zst',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/webkit2gtk_4/2.38.1_x86_64/webkit2gtk_4-2.38.1-chromeos-x86_64.tar.zst'
   })
   binary_sha256({
     aarch64: 'd25a0be821cbf2c710539e685268d47bdcde109ed5a18b2202c132b31b341219',
      armv7l: 'd25a0be821cbf2c710539e685268d47bdcde109ed5a18b2202c132b31b341219',
-       i686: '1a5272472c312165616317481b406085a237f7f5748ab284a826af1faf9c541e',
-     x86_64: '36c5c54bc95ede8c00fb0b35361c4c20213ab6129a2712c10f91ff9c08128942'
+       i686: '5c7acadd5fdf295f2590418c09a87cd9ca305b1af34f982fb5a3e03adc78db88',
+     x86_64: '117945f103fe1af58638451841fff222d5b8c699eb65c05215cce6613f765a65'
```
Comment (uberhacker) [inline packages/webkit2gtk_4.rb:27]: What is wrong with the arm builds?

File: packages/webkit2gtk_4.rb [hunk 9]
```diff
@@ -1,133 +1,180 @@
...
-  @_ver = '2.32.4'
-  version @_ver
+  case ARCH
+  when 'x86_64'
+    version '2.38.1'
+  when 'i686'
+    version '2.38.0-1'
+  when 'aarch64', 'armv7l'
+    version '2.32.4'
+  end
```
Comment (uberhacker) [inline packages/webkit2gtk_4.rb:11]: I would prefer if we didn't do this..

File: packages/webkit2gtk_4_1.rb [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+  description 'Web content engine for GTK'
+  homepage 'https://webkitgtk.org'
+  case ARCH
+  when 'x86_64'
+    version '2.38.1'
+  when 'i686'
+    version '2.38.0'
+  when 'aarch64', 'armv7l'
+    version '2.38.1'
+  end
```
Comment (uberhacker) [inline packages/webkit2gtk_4_1.rb:11]: This is not preferable.

File: packages/webkit2gtk_4_1.rb [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+  source_url 'https://webkitgtk.org/releases/webkitgtk-2.38.1.tar.xz'
+  source_sha256 '02e195b3fb9e057743b3364ee7f1eec13f71614226849544c07c32a73b8f1848'
+
+  binary_url({
+    i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/webkit2gtk_4_1/2.38.0_i686/webkit2gtk_4_1-2.38.0-chromeos-i686.tar.zst',
+  x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/webkit2gtk_4_1/2.38.1_x86_64/webkit2gtk_4_1-2.38.1-chromeos-x86_64.tar.zst'
+  })
+  binary_sha256({
+    i686: 'aad673c3561f317633ab7e1910cc65d9df4b6f32cf1ff43e5930328fc96bb142',
+  x86_64: '6bbadb05897a4a8566eecd6ff7cba5b68d59fe852f1748001cb34b8f726c123d'
```
Comment (uberhacker) [inline packages/webkit2gtk_4_1.rb:23]: Missing arm binaries?

File: packages/webkit2gtk_5.rb [hunk 12]
```diff
@@ -1,101 +1,180 @@
...
-  @_ver = '2.32.4'
-  version @_ver
+  case ARCH
+  when 'x86_64'
+    version '2.38.1'
+  when 'i686'
+    version '2.38.0'
+  when 'aarch64', 'armv7l'
+    version '2.32.4'
+  end
```
Comment (uberhacker) [inline packages/webkit2gtk_5.rb:11]: This is not preferable.

**Review by uberhacker — CHANGES_REQUESTED** [review]
Bump curl version to 7.86.0 for the curl fake package?

File: packages/webkit2gtk_5.rb (hunk 12 above)
Comment (satmandu) [reply packages/webkit2gtk_5.rb:11]: What would you suggest? Just taking out the binaries section for the other architectures?

File: packages/mpv.rb (hunk 6 above)
Comment (satmandu) [reply packages/mpv.rb]: Fixed.

File: packages/webkit2gtk_4.rb (hunk 8 above)
Comment (saltedcoffii) [reply packages/webkit2gtk_4.rb:20]: have you checked around to see if maybe fedora, ubuntu or arch linux have a patch for arm32?

File: packages/webkit2gtk_4.rb (hunk 9 above)
Comment (satmandu) [reply packages/webkit2gtk_4.rb]: This was changed.

File: packages/webkit2gtk_4_1.rb (hunk 11 above)
Comment (satmandu) [reply packages/webkit2gtk_4_1.rb:14]: Build issues...

File: packages/webkit2gtk_5.rb (hunk 12 above)
Comment (satmandu) [reply packages/webkit2gtk_5.rb]: This was changed.

File: packages/webkit2gtk_4_1.rb (hunk 10 above)
Comment (satmandu) [reply packages/webkit2gtk_4_1.rb]: Changed.

File: packages/gemacs.rb (hunk 3 above)
Comment (satmandu) [reply packages/gemacs.rb:18]: adjusted compatibility for now

File: packages/evolution_data_server.rb (hunk 2 above)
Comment (satmandu) [reply packages/evolution_data_server.rb:18]: I will build for armv7l if webkit2gtk_4_1 ends up building for armv7l...

File: bin/crew [hunk 13]
```diff
@@ -1188,7 +1188,11 @@ def shrink_dir(dir)
           # Copying in the ThreadPoolExecutor loop fails non-deterministically
           @execfiles.each_line do |execfilecp|
             execfilecp.slice! '.'
+            next if execfilecp.chomp.empty?
+
             execfilecp = dir + execfilecp.chomp
+            next unless File.exist?(execfilecp)
+
```
Comment (uberhacker) [inline bin/crew:1195]: Bump crew version?

Comment (satmandu) [reply bin/crew:1195]: done

**Review by uberhacker — APPROVED** [review]

**Pull request merged**